nshmdb/fault.py

```diff
@@ -0,0 +1,476 @@
+        return np.linalg.norm(self._corners[-1] - self._corners[0])
+
+    @property
+    def bottom_m(self) -> float:
+        """
+        Returns
+        -------
+        float
+            The bottom depth (in metres).
+        """
+        return self._corners[-1, -1]
+
+    @property
+    def width(self) -> float:
+        """
+        Returns
+        -------
+        float
+            The width of the fault plane (in kilometres).
+        """
+        return self.width_m / _KM_TO_M
+
+    @property
+    def length(self) -> float:
+        """
+        Returns
+        -------
+        float
+            The length of the fault plane (in kilometres).
+        """
+        return self.length_m / _KM_TO_M
+
+    @property
+    def projected_width_m(self) -> float:
+        """
+        Returns
+        -------
+        float
+            The projected width of the fault plane (in metres).
+        """
+        return self.length_m * np.cos(np.radians(self.dip))
+
+    @property
+    def projected_width(self) -> float:
+        """
+        Returns
+        -------
+        float
+            The projected width of the fault plane (in kilometres).
+        """
+        return self.projected_width / _KM_TO_M
+
+    @property
+    def strike(self) -> float:
+        """
+        Returns
+        -------
+        float
+            The bearing of the strike direction of the fault
+            (from north; in degrees)
+        """
+
+        north_direction = np.array([1, 0, 0])
+        up_direction = np.array([0, 0, 1])
+        strike_direction = self._corners[1] - self._corners[0]
+        return qcore.geo.oriented_bearing_wrt_normal(
+            north_direction, strike_direction, up_direction
+        )
+
+    @property
+    def dip_dir(self) -> float:
+        """
+        Returns
+        -------
+        float
+            The bearing of the dip direction (from north; in degrees).
+        """
+        if np.isclose(self.dip, 90):
+            return 0  # TODO: Is this right for this case?
+        north_direction = np.array([1, 0, 0])
+        up_direction = np.array([0, 0, 1])
+        dip_direction = self._corners[-1] - self._corners[0]
+        dip_direction[-1] = 0
+        return qcore.geo.oriented_bearing_wrt_normal(
+            north_direction, dip_direction, up_direction
+        )
+
+    @property
+    def dip(self) -> float:
+        """
+        Returns
+        -------
+        float
+            The dip angle of the fault.
+        """
+        return np.degrees(np.arcsin(np.abs(self.bottom_m) / self.width_m))
+
+    def plane_coordinates_to_global_coordinates(
+        self, plane_coordinates: np.ndarray
+    ) -> np.ndarray:
+        """Convert plane coordinates to nztm global coordinates.
+
+        Parameters
+        ----------
+        plane_coordinates : np.ndarray
+            Plane coordinates to convert. Plane coordinates are
+            2D coordinates (x, y) given for a fault plane (a plane), where x
```

Comment:
Just to be consistent, I think we usually describe it as "along strike" and "down dip"

Reply:
Easy fix, will update.